datalab/datalab.sh

```diff
@@ -16,7 +16,17 @@
 # This init script installs Google Cloud Datalab on the master node of a
 # Dataproc cluster.
 
-set -e -x
+set -exo pipefail
+
+readonly ROLE="$(/usr/share/google/get_metadata_value attributes/dataproc-role)"
+readonly PROJECT="$(/usr/share/google/get_metadata_value ../project/project-id)"
+readonly SPARK_PACKAGES="$(/usr/share/google/get_metadata_value attributes/spark-packages || true)"
+readonly SPARK_CONF='/etc/spark/conf/spark-defaults.conf'
+readonly DATALAB_DIR="${HOME}/datalab"
+readonly PYTHONPATH="/env/python:$(find /usr/lib/spark/python/lib -name '*.zip' | paste -sd:)"
+
+DOCKER_IMAGE="$(/usr/share/google/get_metadata_value attributes/docker-image || true)"
```

Comment:
You can make this also `readonly` if you make this something like `DOCKER_IMAGE=$(...get_metadata_value... || echo 'gcr.io/cloud-datalab/datalab:local')`. And you can remove the logic later in this script.

Reply:
Good point